### ISSUE TITLE
Installation error Failed building wheel for pycatch22

### ISSUE THREAD

**Issue by bcbqwe** [sktime/sktime#5675]
As of Dec 29, 2023
Attempted to install sktime using instructions per:
https://www.sktime.net/en/stable/installation.html
pip install sktime[all_extras]

Please see attached output log.
[sktime-install-error.txt](https://github.com/sktime/sktime/files/13794874/sktime-install-error.txt)

**Comment by yarnabrina** [comment]
Hi @bcbqwe, thanks for the report. We are aware that some mixups happened related to `pycatch22`, but as far as I am aware it happened for `transformations` extra and not for `all_extras`. It was fixed in #5670.

From your logs, the error looks different and seems to be caused by lack of C/C++ compilers. I am not sure if this is a bug from `sktime`'s perspective, what do you suggest @fkiraly?

In the meanwhile, I'll suggest you to install the recommended tool in the error message if you must use `pycatch22` dependent features, else continue with only the soft dependencies you plan to use.

**Comment by fkiraly** [comment]
Yes, this may be sth regarding C compilers, does not seem to be `sktime` related.

I would also suggest installing minimal dependencies, or one of the task specific sets like `forecasting`.
